File: Infrastructure-Module/src/main/kotlin/com/bamyanggang/infrastructuremodule/persistence/UserRepositoryImpl.java
package com.bamyanggang.infrastructuremodule.persistence;

import com.bamyanggang.domainmodule.domain.user.aggregate.User;
import com.bamyanggang.domainmodule.domain.user.repository.UserRepository;
import com.bamyanggang.infrastructuremodule.persistence.user.jpa.entity.UserJpaEntity;
import com.bamyanggang.infrastructuremodule.persistence.user.jpa.repository.UserJpaRepository;
import com.bamyanggang.infrastructuremodule.persistence.user.mapper.UserMapper;
import lombok.AllArgsConstructor;
import org.jetbrains.annotations.NotNull;
import org.springframework.stereotype.Repository;

@Repository
@AllArgsConstructor
public class UserRepositoryImpl implements UserRepository {

    private final UserJpaRepository userJpaRepository;
    private final UserMapper userMapper;

    @Override
    public void save(@NotNull User user) {
        UserJpaEntity userJpaEntity = userMapper.toJpaEntity(user);
        userJpaRepository.save(userJpaEntity);
    }

}

File: Infrastructure-Module/src/main/java/com/bamyanggang/infrastructuremodule/persistence/user/UserRepositoryImpl.kt
package com.bamyanggang.infrastructuremodule.persistence.user

import com.bamyanggang.domainmodule.domain.user.aggregate.User
import com.bamyanggang.domainmodule.domain.user.repository.UserRepository
import com.bamyanggang.infrastructuremodule.persistence.user.jpa.repository.UserJpaRepository
import com.bamyanggang.infrastructuremodule.persistence.user.mapper.UserMapper
import org.springframework.stereotype.Repository

@Repository
class UserRepositoryImpl(
    private val userJpaRepository: UserJpaRepository,
    private val userMapper: UserMapper
) : UserRepository {
    override fun save(user: User) {
        TODO("Not yet implemented")
    }

    override fun findBySocialId(socialId: String): User {
        TODO("Not yet implemented")
    }

}